ie/source/util.h

```diff
@@ -69,13 +69,28 @@ static wstring wstring_replace(const wstring& s, wchar_t from, wchar_t to) {
 
 static wstring wstring_limit(const wstring& s, size_t maxlen = 160) {
     size_t len = s.length();
-    if (len < maxlen) {
+    if (len <= maxlen) {
         return s;
     }
     return s.substr(0, maxlen / 2) +  L" ... <schnip /> ... " + 
         s.substr(len - maxlen / 2);
 }
 
+static wstring wstring_mask_password(const wstring& key, const wstring& value) {
+	// mask sensitive information
+	wstring maskedValue = L"";
+	wstring::size_type isPasskey = key.find(L"passkey");
+	wstring::size_type isAccessToken = key.find(L"access_token");
+	if (isPasskey != wstring::npos || isAccessToken != wstring::npos)
+	{
+		maskedValue = wstring_limit(value, 4);

```

Comment:
not sure if printing first two letters of the password with " ... <schnip /> ... " and last two letters of the password counts as masking. We should simply output password like "***"

Reply:
passkey could be null, that why I choose number 4, for the log to show: null (no quotes).
In other cases, the log content will look like this: "P ... <schnip /> ... 1" for PingOne1
I want to keep the last character for testing purpose if I have correct passkey.